src/smt/theory_str_noodler/formula_preprocess.cpp

```diff
@@ -319,29 +319,39 @@ namespace smt::noodler {
 
             assert(pr.second.get_left_side().size() == 1);
 
-            bool contains_disallowed = !set_disjoint(disallowed_vars, pr.second.get_vars());
-            if (contains_disallowed) {
+            // if right side contains multiple len vars we must do splitting => cannot remove (we can remove if we have only one length var with possibly literals)
+            bool is_right_side_len = !set_disjoint(this->len_variables, pr.second.get_side_vars(Predicate::EquationSideType::Right));
+            if(is_right_side_len && pr.second.get_side_vars(Predicate::EquationSideType::Right).size()) {
```

Comment:
I think I found the source of degradation. There should be `pr.second.get_side_vars(Predicate::EquationSideType::Right).size() > 1`

Reply:
Oh, yeah, that is a stupid bug.

Reply:
Thanks for finding it, I fixed it.